app/controllers/users/dossiers_controller.rb

```diff
@@ -120,6 +122,26 @@ def attestation
       end
     end
 
+    def show_in_trash
+      @hidden_dossier = hidden_dossier_for(params[:id])
+
+      if @hidden_dossier
+        render :show_in_trash
+      else
+        raise ActiveRecord::RecordNotFound
+      end
+    end
+
+    def show_deleted
+      @deleted_dossier = deleted_dossier_for(params[:id])
+
+      if @deleted_dossier
+        render :show_deleted
+      else
+        raise ActiveRecord::RecordNotFound
+      end
+    end
```

Comment:
on peut encore simplifier

```suggestion
    def show_in_trash
      @hidden_dossier = hidden_dossier_for(params[:id])

      raise ActiveRecord::RecordNotFound if @hidden_dossier.nil?
    end

    def show_deleted
      @deleted_dossier = deleted_dossier_for(params[:id])

      raise ActiveRecord::RecordNotFound if @deleted_dossier.nil?
    end
```